src/cli_common.c

```diff
@@ -424,3 +424,21 @@ sds cliVersion(void) {
     }
     return version;
 }
+
+/* This is a wrapper to call redisConnect or redisConnectWithTimeout. */
+redisContext *redisConnectWrapper(const char *ip, int port, const struct timeval tv) {
+    if (tv.tv_sec == 0 && tv.tv_usec == 0) {
+        return redisConnect(ip, port);
+    } else {
+        return redisConnectWithTimeout(ip, port, tv);
+    }
+}
+
+/* This is a wrapper to call redisConnectUnix or redisConnectUnixWithTimeout. */
+redisContext *redisConnectUnixWrapper(const char *path, const struct timeval tv) {
+    if (tv.tv_sec == 0 && tv.tv_usec == 0) {
+        return redisConnectUnix(path);
+    } else {
+        return redisConnectUnixWithTimeout(path, tv);
+    }
+}
```

Comment:
(Not important) Instead of these 4 combinations, we could use redisConnectWithOptions(), like hiredis does internally. Pseudo code:

```C
    redisOptions options = {0};
    if (...) REDIS_OPTIONS_SET_UNIX(&options, path);
    if (...) options.connect_timeout = &tv;
    return redisConnectWithOptions(&options);
```